### PR TITLE
Retry child classes of retried exceptions

### LINKED ISSUE #40: `exceptions_to_retry` does not retry child exceptions
Hi, I'm running into an issue with retrying exceptions. I've added the following configuration to Rspec:

```ruby
config.exceptions_to_retry = [Selenium::WebDriver::Error::WebDriverError]
```

The intention was to catch a whole bunch of Selenium errors that all descend from `WebDriverError`. (A biggy is `Selenium::WebDriver::Error::StaleElementReferenceError`). My expectation was that child exceptions would be caught and retried, similar to how Ruby's `rescue` works.

Unfortunately, it appears that only specific error classes are retried, as indicated by this line:

https://github.com/NoRedInk/rspec-retry/blob/master/lib/rspec/retry.rb#L57

I propose replacing it with something like:

```
break unless exceptions_to_retry.any? do |exception_klass|
  example.exception.class < exception_klass
end
```

### PR DESCRIPTION
**Pull request by dwbutler**
Fixes #40.

### CODE REVIEW

File: lib/rspec/retry.rb [hunk 1]
```diff
@@ -54,7 +54,9 @@ def self.apply
             break if example.exception.nil?
 
             if exceptions_to_retry.any?
-              break unless exceptions_to_retry.include?(example.exception.class)
+              break unless exceptions_to_retry.any? do |exception_klass|
+                example.exception.class <= exception_klass
```
Comment (michaelglass) [inline lib/rspec/retry.rb]: would prefer `example.exception.is_a? exception_klass` but will get this merged either way

File: spec/lib/rspec/retry_spec.rb [hunk 2]
```diff
@@ -77,9 +77,16 @@ def shift_expectation
...
-    describe "with a list of exceptions", :retry => 2, :exceptions_to_retry => [NoMethodError] do
+    describe "with a list of exceptions", :retry => 2, :exceptions_to_retry => [NameError] do
       context "the example throws an exception contained in the retry list" do
         it "retries the maximum number of times" do
+          raise NameError unless count > 1
+          expect(count).to eq(2)
+        end
+      end
+
+      context "the example throws a child of an exception contained in the retry list" do
```
Comment (michaelglass) [inline spec/lib/rspec/retry_spec.rb]: nice test

File: lib/rspec/retry.rb (hunk 1 above)
Comment (dwbutler) [inline lib/rspec/retry.rb]: That's a good idea, and definitely more clear. I'll push that.

File: spec/lib/rspec/retry_spec.rb (hunk 2 above)
Comment (dwbutler) [inline spec/lib/rspec/retry_spec.rb]: Thanks!

**Pull request merged**